connector/src/main/java/org/geysermc/connector/utils/SettingsUtils.java

```diff
@@ -57,11 +57,24 @@ public static void buildForm(GeyserSession session) {
         CustomFormBuilder builder = new CustomFormBuilder(LanguageUtils.getPlayerLocaleString("geyser.settings.title.main", language));
         builder.setIcon(new FormImage(FormImage.FormImageType.PATH, "textures/ui/settings_glyph_color_2x.png"));
 
-        // Client can only see its coordinates if reducedDebugInfo is disabled and coordinates are enabled in geyser config.
-        if (!session.isReducedDebugInfo() && session.getConnector().getConfig().isShowCoordinates()) {
+        // Only show the client title if any of the client settings are available
+        if (session.getPreferencesCache().isAllowShowCoordinates() || !(CooldownUtils.getDefaultShowCooldown() == CooldownUtils.CooldownType.DISABLED)) {
             builder.addComponent(new LabelComponent(LanguageUtils.getPlayerLocaleString("geyser.settings.title.client", language)));
 
-            builder.addComponent(new ToggleComponent(LanguageUtils.getPlayerLocaleString("geyser.settings.option.coordinates", language), session.getWorldCache().isPrefersShowCoordinates()));
+            // Client can only see its coordinates if reducedDebugInfo is disabled and coordinates are enabled in geyser config.
+            if (session.getPreferencesCache().isAllowShowCoordinates()) {
+                builder.addComponent(new ToggleComponent(LanguageUtils.getPlayerLocaleString("geyser.settings.option.coordinates", language), session.getPreferencesCache().isPrefersShowCoordinates()));
+            }
+
+            if (!(CooldownUtils.getDefaultShowCooldown() == CooldownUtils.CooldownType.DISABLED)) {
```

Comment:
Just use `CooldownUtils.getDefaultShowCooldown() != CooldownUtils.CooldownType.DISABLED`.

Reply:
addressed in 8767aacf545324b095e599bb307afe51ac7d0ea1